modules/core/src/encoder/SemanticPreEncoder.ts

```diff
@@ -41,8 +41,12 @@ export class SemanticPreEncoder {
           if (value) {
 
             const vendor = gvl.vendors[vendorId];
+            const isInactive = !vendor || vendor.deletedDate;
+            const isSpecialPurposesOnly = !isInactive
```

Comment:
nit: can simplify (remove the conditional) by using `&&` instead of ` ? ... : false`

Reply:
To both minimize the scope of this variable and simplify it by not having to worry about inactive vendors, I'd compute this in the `else` block of `if(isInactive)` (changing the `else if` on #L54 to a full `else`). It's a lot more re-indenting, but I think it keeps the logic properly contained where we want it.

Reply:
I can simplify this even more by stating that [a vendor who has a `deletedDate` is not included in the list of vendors ](https://github.com/InteractiveAdvertisingBureau/iabtcf-es/blob/81d39d2df895bb567a096c733f07cf5f73bac974/modules/core/src/GVL.ts#L596)when the `GVL` class is created.  So `vendor.deletedDate` will not be there.

Reply:
Thanks for the update, Juan. No need for the const var, eh?
Just...

```
              if (gvlVendorKey === 'legIntPurposes' && (vendor['purposes'].length === 0 && vendor['legIntPurposes'].length === 0 && vendor['specialPurposes'].length > 0)) {

                /**
                 * Per June 2021 Policy change, Vendors declaring only Special Purposes may have their legitimate interest Vendor bit set if they have been disclosed. 
                 * This empty block ensures their LI bit remains set
                 */

              } else {
              ......
```

Reply:
Yes, I keep the const to not have a big if condition but if you prefer I can avoid the const. Just let me know I will do the update.

Thank you for the review

Reply:
Updated and added the new comment